### ISSUE TITLE
MagicNumber rule does not trigger for unsigned literals

### ISSUE THREAD

**Issue by YSakhno** [detekt/detekt#8194]
## Expected Behavior

The `MagicNumber` rule should be reporting 'magic numbers' written as unsigned number literals (for example, `65520U`).

## Observed Behavior

If my code contains a 'magic number' written as an unsigned literal, the rule does not trigger. Example:

```kotlin
    a = 65520U // does not trigger
    b = 0xFFF0U // also 'fine'
    c = 65520.toUInt() // now it triggers
    d = 0xFFF0.toUInt() // triggers on this one too
```

## Steps to Reproduce

See **Observed Behavior** above.

## Context

N/A (should be obvious from the rule's purpose)

## Your Environment

* Version of detekt used: 1.23.8
* Version of Gradle used (if applicable): 8.14.1
* Operating System and version: Windows 11 Pro 24H2

My `detekt.yaml` has this:

```yaml
  MagicNumber:
    ignoreCompanionObjectPropertyDeclaration: false
    ignoreNamedArgument: false
    ignoreExtensionFunctions: false
```

**Comment by BraisGabin** [comment]
You are right. Do you want to give it a try a send a pr fixing it?

**Comment by YSakhno** [comment]
I can try, but no promises

**Comment by CziSKY** [comment]
@YSakhno Hi, it seems to have been inactive for a couple of weeks. I believe I've identified the root cause and have a fix in mind.
Would it be okay if I pick this up?
